new/ohai-config.md

```diff
@@ -0,0 +1,61 @@
+---
+RFC: unassigned
+Author: Bryan McLellan <btm@loftninjas.org>
+Status: Draft
+Type: Standards Track
+---
+
+# Ohai User Configuration
+
+Provides a standard for passing configuration settings to Ohai to configure its behavior as well as that of plugins.
+
+## Motivation
+
+    As a Chef user,
+    I want to easily configure Ohai,
+    so that it behaves optimally on my infrastructure.
+
+    As an Ohai user,
+    I want Ohai to load configuration settings from my client.rb,
+    so that it behaves the same as it does during a chef-client run.
+
+## Specification
+
+### Configuration File
+
+Previously Ohai has not had a configuration file. A small number of Ohai configuration settings do exist that can be specified in the chef-client ```client.rb``` file, which are then passed to Ohai during a chef-client run. For example, additional locations on the file system can be searched for plugins by adding paths to the ```Ohai::Config[:plugin_path]``` array. This file is only loaded by Chef and does not affect Ohai when being run on the command line nor any other programs loading Ohai as a library.
+
+When run from the command line, Ohai should load the ```client.rb``` file from the appropriate platform specific path unless an alternate configuration file is provided as a command line argument. To preserve compatibility with the existing parsing of the ```client.rb```, the Ohai configuration file syntax will be variables set on the ```Ohai::Config``` class,, and barewords will be ignored using method_missing.
+
+For example:
+
+```
+Ohai::Config[:plugin_path] << "/usr/local/lib/ohai/plugins"
+Ohai::Config[:plugin][:hostname][:fqdn_using] = [ :hostname, :nis, :dns ]
+
+# This is ignored:
+node_name "darius"
+```
+
+Using the same ```client.rb``` file used by the chef-client also allows both tools to be configured simultaneously by populating this file during bootstrap with custom local values (when that is supported in the future).
+
+### Namespacing
+
+To reduce the risk of built-in and custom plugins using the same configuration setting for conflicting purposes, plugins must prefix their configuration settings with ```[:plugin]``` and the name of the plugin consuming the setting. 
```

Comment:
Hm... I can argue a few things, and I think this comes from not fully understanding or knowing what kinds of configuration would be set here.

Arguments for namespacing with the plugin's `:Name`: 
1. specific application of configuration settings to one plugin won't affect others which provide overlapping attributes
2. potentially replace hints files
3. write less, configure more (Lamont's last point)

Argument for namespacing with the attribute's `[:possibly][:chained][:name]`:
A. configure attributes provided across plugins
B. explicitly set attribute value (not sure if this would be for pre-population or values added after ohai runs)
C. no need to know which plugin provides which attributes

I think C is pretty important because Ohai 7 kind of did away with the relationship between a plugin and an attribute. Unless you're a plugin author or don't mind searching through ohai source code, you probably don't know or care which plugin provides which attribute.

A more complicated (and potentially more confusing to users and harder to reason about and debug) option would be to harness the syntactic requirements already in ohai. Plugins are declared by symbols starting with capital letters while attributes are lower case strings. 

```ruby
Ohai::Config[:Foo][:bar] = value    # Set the 'bar' attribute of the Foo plugin to value
Ohai::Config[:bar] = value             # Set the 'bar' attribute to value
```

This would have to be /extremely/ well documented.